### ISSUE TITLE
Kustomize version updating to latest patchset of minor 3.8

### ISSUE THREAD

**Issue by Fixmetal** [argoproj/argo-cd#5297]
# Summary

ArgoCD 1.8 contains an "old" kustomize version which is 3.8.1. As of today kustomize 3.8 is at 3.8.9.

# Motivation

Kustomize versions up to 3.8.2 are bugged by this https://github.com/kubernetes-sigs/kustomize/issues/2960. Seems `3.8.3` has been patched. See relevant PRs for KyYAML and API:

- https://github.com/kubernetes-sigs/kustomize/pull/2975
- https://github.com/kubernetes-sigs/kustomize/issues/2960

Release notes:
- https://github.com/kubernetes-sigs/kustomize/releases/tag/kustomize%2Fv3.8.3
- https://github.com/kubernetes-sigs/kustomize/releases/tag/api%2Fv0.6.1
- https://github.com/kubernetes-sigs/kustomize/releases/tag/kyaml%2Fv0.8.0

# Proposal

It would be good if ArgoCD 1.8 would suit the latest minor release of kustomize 3.8 which is 3.8.9 as of today

**Comment by ainmosni** [comment]
Yeah I hit this bug yesterday as well and worked around it by installing this kustomize version but it would be nice if that wasn't needed for maintenance reasons. I submitted a PR that fixes this.

**Comment by qrilka** [comment]
Also the current `kustomize` isn't working correctly with backend service names for ingress in k8s-1.19 and later (it's now service/name instead of seviceName)  as https://github.com/kubernetes-sigs/kustomize/commit/62e4df72d32f615dd88e78bdb3654f043b9d5b44 appeared only in the version 3.8.6
